### ISSUE TITLE
Importing 3rd party svelte components doesn't work

### ISSUE THREAD

**Issue by cdock1029** [jakedeichert/svelvet#47]
Tying to use `svelte-spa-router` and seeing this error on compile simply by importing the library:

`import Router from 'svelte-spa-router`

```
Failed to build with snowpack
- snowpack installing... 
Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
    at error (/home/cdock/Documents/waldon-app/svelte-app/node_modules/rollup/dist/shared/node-entry.js:5400:30)
    at Module.error (/home/cdock/Documents/waldon-app/svelte-app/node_modules/rollup/dist/shared/node-entry.js:9820:16)
    at tryParse (/home/cdock/Documents/waldon-app/svelte-app/node_modules/rollup/dist/shared/node-entry.js:9713:23)
    at Module.setSource (/home/cdock/Documents/waldon-app/svelte-app/node_modules/rollup/dist/shared/node-entry.js:10076:33)
    at Promise.resolve.catch.then.then.then (/home/cdock/Documents/waldon-app/svelte-app/node_modules/rollup/dist/shared/node-entry.js:12362:20)

```
Are third-party svelte components not use-able currently?

**Comment by jakedeichert** [comment]
I see `svelte-spa-router` exports `Router.svelte` uncompiled. Is that normal?

I thought it's more common that libs expose their built/dist js versions 🤔 

So to answer your question, svelvet and snowpack **do not** touch any 3rd party dependencies at all. So importing svelte components that are not in your project is currently not supported.

Two possible workarounds:

1. Copy their Router into your project so it gets compiled via svelvet. 

2. Ask `svelte-spa-router` to include a dist/built js version of their `Router`.

**Comment by dangodev** [comment]
I also experienced the same issue with `svelte-routing`. You’re correct—Snowpack doesn’t transpile third-party deps, and the error message comes from Rollup trying to bundle a `*.svelte` file it doesn’t understand. However, it looks like a few Svelte third-party packages ship code this way! So it might be hard to ask them all to ship transpiled code.

I’m going to see if there’s a workaround in Snowpack without rolling in the Svelte compiler as a dependency of Snowpack 🙂

**Comment by jakedeichert** [comment]
> However, it looks like a few Svelte third-party packages ship code this way!

Ahh! Interesting... I haven't noticed how common it is. I think i understand why now.. that means `svelte-routing` and other libs would need to depend on a specific version of svelte and precompile using that which could break on the user's side if svelte changes behaviour in minor/patch versions?

Yeah, this needs to be fixed/supported on svelvet's side I think. 

One problem I foresee with `svelte-routing` is that they have an `index.js` which exports the svelte files like so:

```js
export { default as Router } from "./Router.svelte";
export { default as Route } from "./Route.svelte";
export { default as Link } from "./Link.svelte";
```

For svelvet to determine **when** to compile a dependency, I think we'd need to force people to import them individually using the `.svelte` extension.

Instead of:

```js
  import { Router, Link, Route } from "svelte-routing";
```

You'd have to do:

```js
  import Link from "svelte-routing/src/Link.svelte";
  import Route from "svelte-routing/src/Route.svelte";
  import Router from "svelte-routing/src/Router.svelte";
```

That's what I'm thinking so far at least.

**Comment by FredKSchott** [comment]
Hey yall, just to share: @dangodev started a conversation on the Snowpack discussion board as well for this one: https://www.pika.dev/npm/snowpack/discuss/79

**Comment by dangodev** [comment]
I just merged a [PR](https://github.com/pikapkg/snowpack/pull/211) in Snowpack that allows one to specify custom Rollup plugins. With a local build of Snowpack (not yet released to npm), I was able to use third-party Svelte components in my project by running:

```bash
npm install rollup-plugin-svelte
```

And adding a `snowpack.config.js` to the root of my project with the following:

```js
const rollupPluginSvelte = require('rollup-plugin-svelte');

module.exports = {
  rollup: {
    plugins: [rollupPluginSvelte()]
  }
};
```

Note that `snowpack.config.js` lives in my **local project**, not Svelvet itself. Because when Svelvet runs, the Snowpack is executing from `/[local-project]/node_modules/snowpack`, it’s only able to find `/[local-project]/snowpack.config.js`.

Anyway, this is a workaround that seems to fix this problem with Snowpack’s next npm release. But maybe we can find a way for this to all happen more behind-the-scenes with Svelvet? Open to ideas.

**Comment by jakedeichert** [comment]
@dangodev thanks for making that PR to snowpack! That definitely makes things easier on svelvet's side. And thanks @FredKSchott for merging that... it does seem like valuable feature being able to add rollup plugins for snowpack to use :100:

I'll try this out this week to see if there's anything svelvet can do here to automate this.

**Two ideas so far:**
1. Svelvet could auto create a `snowpack.config.js` file in the user's project dir if it doesn't exist. 
2. We could maybe PR a feature to snowpack where their cli can take a `--config path/to/snowpack.config.js` flag so that svelvet can _own_ the config instead of the user. Thoughts @FredKSchott ?


As for today, I'm going to upgrade snowpack's version in svelvet and make a release so that people can start using a `snowpack.config.js` file for this.

**Comment by FredKSchott** [comment]
Awesome! Yea, config merging is a tricky business, but something that we could look into if needed. We could also do over the command line specific to this use-case, something like `--plugin path/to/plugin.js`. That would hopefully be enough for Svelvet and others like it.

I'll spin up an issue in our repo to discuss!

**Comment by jakedeichert** [comment]
@cdock1029 just released v0.3.2 with the latest version of snowpack 👍 

Just add a [`snowpack.config.js`](https://github.com/jakedeichert/svelvet/blob/master/tests/snapshot-snowpack-config/snowpack.config.js) and `npm i -D rollup-plugin-svelte` like @dangodev showed above!




I'm going to close this issue in favour of #49

**Comment by jakedeichert** [comment]
@dangodev whoops. I edited your comment by accident, thinking it was mine.

I updated the code block to be:

```
const rollupPluginSvelte = require("rollup-plugin-svelte");

module.exports = {
  rollup: {
    plugins: [
      rollupPluginSvelte({
        dev: process.env.NODE_ENV !== "production"
      })
    ]
  }
};
```

Not sure if there's a way for me to undo the edit :/
